### PR TITLE
feat(tooltip): for customHTML pass along the HTML tooltip

### PR DESCRIPTION
**Pull request by scottdickerson**
### Updates
- feat(tooltip-bar) add the passing of the original rendered HTML tooltip rendered to the customHTML function
- feat(tooltip) add the passing of the original rendered HTML tooltip to the customHTML function

### Demo screenshot or recording
image

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/IBM/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

### CODE REVIEW

**Comment by theiliad** [comment]
Closes #376

File: packages/core/src/components/essentials/tooltip.ts [hunk 1]
```diff
@@ -43,15 +43,21 @@ export class Tooltip extends Component {
...
+				let defaultTooltip;
+				if (e.detail.multidata) {
 					// multi tooltip
 					data = e.detail.multidata;
-					tooltipTextContainer.html(this.getMultilineTooltipHTML(data));
+					defaultTooltip = this.getMultilineTooltipHTML(data);
+				} else {
+					defaultTooltip = this.getTooltipHTML(data);
+				}
+				// if there is a provided tooltip HTML function call it
```
Comment (natashadecoste) [inline packages/core/src/components/essentials/tooltip.ts]: suggesting an empty line before the comment (in between the sets of if/else) to maintain consistency with tooltip-bar

File: packages/core/src/components/essentials/tooltip-bar.ts [hunk 2]
```diff
@@ -31,23 +31,25 @@ export class TooltipBar extends Tooltip {
 
 				const hoveredElement = e.detail.hoveredElement.node();
 
-				// if there is a provided tooltip HTML function
-				if (Tools.getProperty(this.model.getOptions(), "tooltip", "customHTML")) {
-					tooltipTextContainer.html(this.model.getOptions().tooltip.customHTML(hoveredElement));
-				} else if (e.detail.multidata) {
+				let defaultTooltip;
```
Comment (theiliad) [inline packages/core/src/components/essentials/tooltip-bar.ts]: let's call this `defaultHTML`

File: packages/core/src/components/essentials/tooltip.ts [hunk 3]
```diff
@@ -43,15 +43,21 @@ export class Tooltip extends Component {
 
 				let data = select(event.target).datum() as any;
 
-				// if there is a provided tooltip HTML function
-				if (Tools.getProperty(this.model.getOptions(), "tooltip", "customHTML")) {
-					tooltipTextContainer.html(this.model.getOptions().tooltip.customHTML(data));
-				} else if (e.detail.multidata) {
+				// Generate default tooltip
+				let defaultTooltip;
```
Comment (theiliad) [inline packages/core/src/components/essentials/tooltip.ts]: `defaultHTML`

**Comment by scottdickerson** [comment]
this now prereqs #390 because I accidentally merged that branch in for some testing.  Please merge that one first

**Pull request merged**